src/NuGet.Core/NuGet.ProjectModel/PackageSpecReferenceDependencyProvider.cs

```diff
@@ -339,8 +339,8 @@ private List<LibraryDependency> GetDependenciesFromSpecRestoreMetadata(PackageSp
 
                 dependencies.AddRange(targetFrameworkInfo.Dependencies);
 
-#if enableCPVMTransitivePinning
-                if (packageSpec.RestoreMetadata?.CentralPackageVersionsEnabled == true)
+                if (packageSpec.RestoreMetadata?.CentralPackageVersionsEnabled == true &&
+                    packageSpec.RestoreMetadata?.TransitiveDependencyPinningEnabled == true)
```

Comment:
Let's get rid of these old comparisons to `true`
```suggestion
                if (packageSpec.RestoreMetadata?.CentralPackageVersionsEnabled &&
                    packageSpec.RestoreMetadata?.TransitiveDependencyPinningEnabled)
```

Reply:
Both terms are nullable so the comparison to `true` is rather necessary.